README.md

```diff
@@ -21,147 +24,131 @@ or to use in an npm script
+* build
 
-### .Nodeshift Directory
+* deploy
 
-The `.nodeshift` directory is responsible for holding your resource files.  These are `.yml` files that describe you service, deployments and routes.
+* undeploy
 
-Currently, nodeshift will only create resources based on the files specified,  in the future, its possible somethings could be created by default
+Options that you can specify on the command line, can also be passed as an options object to the API
 
-#### Commands & Goals
+#### Example Usage
 
-By default, if you run just `nodeshift`, it will run the `deploy` goal, which runs everything.
+    const nodeshift = require('nodeshift');
 
-**resource** - will parse and create the application resources files on disk
+    // Deploy an Application
+    nodeshift.deploy().then((response) => {
+        console.log(response);
+        console.log('Application Deployed')
+    }).catch((err) => {
+        console.log(err);
+    })
 
-**apply-resource** - does the resource goal and will also push that to your running cluster
+_please note: Currently, once a route, service, deployment config, build config, and imagestream config are created, those are re-used. The only thing that changes from deployment to deployment is the source code.  For application resources, you can update them by undeploying and then deploying again.  BuildConfigs and Imagestreams can be re-created using the --build.recreate flag_
 
-**deploy** - does everything, the archiving of code and everything that apply-resource does
+## Advanced Options
 
-**undeploy** - removes things that were deployed with the resource command
+While nodeshift tries to be very opinionated during the deployment process, but there are options available to pass to the cli or the API.
 
-**build** - archives the code, creates build config and imagestream and pushes the binary to the cluster
+#### version
+Outputs the current version of nodeshift
 
-#### Template Parameters
+#### projectLocation
+Changes the default location of where to look for your project. Defaults to your current working directory(CWD)
 
-Some templates might need to have a value set at "run time".  For example, in the template below, we have the `${SSO_AUTH_SERVER_URL}` parameter:
+#### configLocation
+Changes the default location of where to look for a configfile. Defaults to ~/.kube/
 
-        apiVersion: v1
-        kind: Deployment
-        metadata:
-            name: nodejs-rest-http-secured
-        spec:
-          template:
-            spec:
-              containers:
-                - env:
-                  - name: SSO_AUTH_SERVER_URL
-                    value: "${SSO_AUTH_SERVER_URL}"
-                  - name: REALM
-                    value: master
+#### osc.strictSSL
+Setting to pass to the [Openshift Rest Client](https://www.npmjs.com/package/openshift-rest-client) for SSL use.  To allow using a self-signed cert, set to false
```

Comment:
Maybe here and in `osl.tryServiceAccount` the wording could be a little clearer. It took me a few reads to understand what you mean. How about "This option is passed through to the ...".